saleor/order/utils.py

```diff
@@ -909,123 +888,124 @@ def update_discount_for_order_line(
         )
-
-        order_line.unit_discount = undiscounted_base_unit_price - base_unit_price
-
-        order_line.unit_price = TaxedMoney(base_unit_price, base_unit_price)
         order_line.base_unit_price = base_unit_price
 
-        order_line.unit_discount_type = value_type
-        order_line.unit_discount_value = value
-        # TODO: should we save those values?
-        order_line.total_price = order_line.unit_price * order_line.quantity
-        order_line.undiscounted_unit_price = (
-            order_line.unit_price + order_line.unit_discount
-        )
-        order_line.undiscounted_total_price = (
-            order_line.quantity * order_line.undiscounted_unit_price
-        )
-        fields_to_update.extend(
-            [
-                "tax_rate",
-                "unit_discount_value",
-                "unit_discount_amount",
-                "unit_discount_type",
-                "unit_discount_reason",
-                "unit_price_gross_amount",
-                "unit_price_net_amount",
-                "total_price_net_amount",
-                "total_price_gross_amount",
-                "base_unit_price_amount",
-                "undiscounted_unit_price_gross_amount",
-                "undiscounted_unit_price_net_amount",
-                "undiscounted_total_price_gross_amount",
-                "undiscounted_total_price_net_amount",
-            ]
-        )
+        update_unit_discount_data_on_order_line(order_line, line_discounts)
+        fields_to_update = [
+            "unit_discount_value",
+            "unit_discount_amount",
+            "unit_discount_type",
+            "unit_discount_reason",
+            "base_unit_price_amount",
+        ]
+        order_line.save(update_fields=fields_to_update)
 
-    # Save lines before calculating the taxes as some plugin can fetch all order data
-    # from db
-    order_line.save(update_fields=fields_to_update)
 
-    _update_manual_order_line_discount_object(
-        value, value_type, reason, order_line, order.currency
-    )
+def _remove_invalid_discounts_for_adding_manual(
+    order_line_discounts: list[OrderLineDiscount],
+):
+    """Remove all line discounts except the single manual line discount."""
+    discount_to_delete = []
+    current_manual_discount = None
+    for discount in order_line_discounts:
+        if discount.type == DiscountType.MANUAL and not current_manual_discount:
+            current_manual_discount = discount
+        else:
+            discount_to_delete.append(discount)
 
+    if discount_to_delete:
+        for discount in discount_to_delete:
+            order_line_discounts.remove(discount)
+        OrderLineDiscount.objects.filter(
+            id__in=[discount.id for discount in discount_to_delete]
+        ).delete()
 
-def _update_manual_order_line_discount_object(
-    value, value_type, reason, order_line, currency
-):
+
+def _get_order_line_discount_to_update(
```

Comment:
suggestion (non-blocking): Seems like the function should be named `_get_order_line_manual_discount`
as we basically looking for manual discount